pkg/controller/containersource/reconcile.go

```diff
@@ -64,13 +66,13 @@ func (r *reconciler) Reconcile(ctx context.Context, object runtime.Object) (runt
 	// See if the source has been deleted
 	accessor, err := meta.Accessor(source)
 	if err != nil {
-		logger.Warnf("Failed to get metadata accessor: %s", err)
+		logger.Warnf("Failed to get metadata accessor: %s", zap.Error(err))
 		return object, err
 	}
 	// No need to reconcile if the source has been marked for deletion.
 	deletionTimestamp := accessor.GetDeletionTimestamp()
 	if deletionTimestamp != nil {
```

Comment:
Is there no plans to use finalizers? Don't we need to run some code to remove the source at some point?

Reply:
As in the old sources used start/stop model, and in those cases you'd run some code to actually undo what the subscription did.

Reply:
I will test this but I think because I am owning the deployment, the deployment will be deleted if the source is.

Reply:
Cascade delete works.

Reply:
If the deployment creates external resources, how will it know to delete those external resources when the source is deleted? I don't think it can use termination signals, because those might be generated by the node lifecycle.

We could punt on this by saying that the container source won't be able to manage external resources.

Reply:
Right, what @grantr says is what I tried to say. It's not enough to just kill the deployment, there needs to be a way to say, please undo the "subscribe" that you did before. Kind of like what the previous sources did, they kept enough of the context around that was passed back during unsubscribe.

Reply:
OOHHHH!! I see. yes. This is a good idea. I made https://github.com/knative/eventing-sources/issues/21 to solve it.